DependencyInjection/Configuration.php

```diff
@@ -473,22 +473,52 @@ private function addExceptionSection(ArrayNodeDefinition $rootNode)
                     ->addDefaultsIfNotSet()
                     ->canBeEnabled()
                     ->children()
+                        ->booleanNode('map_exception_codes')
+                            ->defaultFalse()
+                            ->info('Enables an event listener that maps exception codes to response status codes based on the map configured with the "fos_rest.exception.codes" option.')
+                        ->end()
+                        ->booleanNode('exception_listener')
+                            ->defaultValue(function () {
+                                @trigger_error('Enabling the "fos_rest.exception.exception_listener" option is deprecated since FOSRestBundle 2.8.', E_USER_DEPRECATED);
+
+                                return true;
+                            })
+                            ->beforeNormalization()
+                                ->ifTrue()
+                                ->then(function ($v) {
+                                    @trigger_error('Enabling the "fos_rest.exception.exception_listener" option is deprecated since FOSRestBundle 2.8.', E_USER_DEPRECATED);
+
+                                    return $v;
+                                })
+                            ->end()
+                        ->end()
                         ->scalarNode('exception_controller')
-                            ->defaultValue(static function () {
-                                @trigger_error('Not setting the "fos_rest.exception.exception_controller" configuration option is deprecated since FOSRestBundle 2.8. Its default value will be set to "fos_rest.exception.controller::showAction" in 3.0.', E_USER_DEPRECATED);
+                            ->defaultNull()
+                            ->setDeprecated('The "%path%.%node%" option is deprecated since FOSRestBundle 2.8.')
+                        ->end()
+                        ->booleanNode('serialize_exceptions')
+                            ->defaultValue(function () {
+                                @trigger_error('Enabling the "fos_rest.exception.serialize_exceptions" option is deprecated since FOSRestBundle 2.8.', E_USER_DEPRECATED);
```

Comment:
Why deprecating this? We using exception code in the api to handle errors in the frontend application. We are required to have the response like: `{"code": "7007", "message": "Exception Message"}` in our rest api.

Reply:
The new `FlattenExceptionHandler`/`FlattenExceptionNormalizer` should handle that (see the dedicated tests). You just need to make sure that exceptions are handled by the ErrorHandler component.

Reply:
Ok that would be great!